src/reconstruction/refinement/addReaction.m

```diff
@@ -258,108 +275,143 @@
-        model.mets{end+1,1} = metaboliteList{i};
-        newMetsCoefs(end+1) = stoichCoeffList(i);
-        if (isfield(model,'metNames'))      %Prompts to add missing info if desired
-            model.metNames{end+1,1} = regexprep(metaboliteList{i},'(\[.+\]) | (\(.+\))','') ;
-            warning(['Metabolite name for ' metaboliteList{i} ' set to ' model.metNames{end}]);
-            %             model.metNames(end) = cellstr(input('Enter complete metabolite name, if available:', 's'));
-        end
-        if (isfield(model,'metFormulas'))
-            model.metFormulas{end+1,1} = '';
-            warning(['Metabolite formula for ' metaboliteList{i} ' set to ''''']);
-            %             model.metFormulas(end) = cellstr(input('Enter metabolite chemical formula, if available:', 's'));
-        end
-        if isfield(model,'metChEBIID')
-            model.metChEBIID{end+1,1} = '';
-        end
-        if isfield(model,'metChEBIID')
-            model.metChEBIID{end+1,1} = ''; %changed to match Recon 2 nomenclature
-        end
-        if isfield(model,'metKEGGID')
-            model.metKEGGID{end+1,1} = '';
-        end
-        if isfield(model,'metPubChemID')
-            model.metPubChemID{end+1,1} = '';
-        end
-        if isfield(model,'metInChIString')
-            model.metInChIString{end+1,1} = '';
-        end        
-        if isfield(model,'metCharges')
-            model.metCharges(end+1,1) = 0;
-        end
-        if isfield(model,'metHepatoNetID')
-            model.metHepatoNetID{end+1,1} = ''; %added
-        end
-        if isfield(model,'metEHMNID')
-            model.metEHMNID{end+1,1} = ''; %added
-        end
-        if isfield(model,'metHMDBID')
-            model.metHMDBID{end+1,1} = ''; %added
-        end
     end
 end
 
-%printLabeledData(model.mets,Scolumn,1);
-
-if isfield(model,'b')
-    model.b = [model.b;zeros(length(model.mets)-length(model.b),1)];
-end
 
 % if ~oldRxnFlag, model.rxnGeneMat(rxnID,:)=0; end
 
-if (isfield(model,'genes'))
-    if (nargin < 11)
-        model = changeGeneAssociation(model,rxnName,grRule);
+if (isfield(model,'genes')) 
+    if isempty(parser.Results.systNameList)        
+        model = changeGeneAssociation(model,rxnID,grRule);
     else
-        %fprintf('In addReaction, the class of systNameList is %s',
-        %class(systNameList)); % commented out by Thierry Mondeel
-        if ~isempty(geneNameList) && ~isempty(systNameList)
-        model = changeGeneAssociation(model,rxnName,grRule,geneNameList,systNameList);
-        else
-            model = changeGeneAssociation(model,rxnName,grRule);
-        end
+        model = changeGeneAssociation(model,rxnID,grRule,geneNameList,systNameList);
     end
 end
 
 % Figure out if the new reaction already exists
 rxnInModel=false;
+duplicatePos = [];
 if (nNewMets > 0) && isempty(find(newMetsCoefs == 0, 1))
-    Stmp = [model.S;sparse(nNewMets,nRxns)];
+    Stmp = model.S;

```

Comment:
I think it was simply a sanity check. you should not look for duplicates, if there are new Metabolites in the reaction.